research-software-directory/RSD-as-a-service
# Fix VSCode project settings for ESLint and formating

* fix/update VSCode project settings and ESLint config to enable consistent formatting
* tested with VSCode v1.81.1 + ESLint plugin v2.4.2

Related to #915

Changes proposed in this pull request:

* removed outdated parameters
* use [Multiple language specific editor settings](https://code.visualstudio.com/updates/v1_63#_multiple-language-specific-editor-settings) introduced in VSCode v1.63.0
* ESLint rules:
  * reactivated `indent` rule to enable auto-formatted indention
  * added option to format "switch" statements
* re-ordered configs (with in files) by topic

How to test:

* create a new js, jsx, ts or tsx file, insert some code and check if correct formatting happens during save file
* edit a js, jsx, ts or tsx file, alter some code and check if correct formatting happens during save file
